modules/gdnavigation/nav_utils.h

```diff
@@ -104,21 +107,17 @@ struct Polygon {
 	Vector3 center;
 };
 
-struct Connection {
-	Polygon *A = nullptr;
-	int A_edge = -1;
-	Polygon *B = nullptr;
-	int B_edge = -1;
-};
-
 struct NavigationPoly {
 	uint32_t self_id = 0;
 	/// This poly.
 	const Polygon *poly;
-	/// The previous navigation poly (id in the `navigation_poly` array).
-	int prev_navigation_poly_id = -1;
-	/// The edge id in this `Poly` to reach the `prev_navigation_poly_id`.
-	uint32_t back_navigation_edge = 0;
+
+	/// Those 4 variables are used to travel the path backwards.
+	int back_navigation_poly_id = -1;
+	uint32_t back_navigation_edge = -1;
```

Comment:
This is an `uint32_t` try to use `UINT32_MAX` instead of `-1`, or use an `int` or an `int64`. Otherwise it's an undefined behavior.